### ISSUE TITLE
[Bug Report] Failed to concatenate observations in vector envs when env is wrapped with NormalizeObservation

### ISSUE THREAD

**Issue by KexianShen** [Farama-Foundation/Gymnasium#415]
### Describe the bug

Failed to concatenate observations in vector envs when env is wrapped with NormalizeObservation

### Code example

```shell
import gymnasium as gym


def make_env():
    def thunk():
        env = gym.make("CarRacing-v2")
        env = gym.wrappers.GrayScaleObservation(env)
        env = gym.wrappers.NormalizeObservation(env)
        return env

    return thunk


if __name__ == "__main__":
    envs = gym.vector.SyncVectorEnv([make_env() for i in range(4)])
    obs, _ = envs.reset()
```

### System info

Describe how Gymnasium was installed: pip
Version of gymnasium: 0.28.1
What OS/version of Linux you're using: Ubuntu 22.04
Python version: 3.9.16

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

**Comment by pseudo-rnd-thoughts** [comment]
It looks like the observation space in `NormalizeObservation` is never updated.
I would recommend in the thunk adding `env.observation_space.dtype = np.float32` which should solve it.

Ironically, the `NormalizeObservation` has never been technically correct. We will look to fix this in the new experimental `NormalizeObservationV0` wrapper.
```python
import pytest

from gymnasium.experimental.wrappers import NormalizeObservationV0
from tests.experimental.wrappers.utils import TESTING_OBS_ENVS_IDS, TESTING_OBS_ENVS


@pytest.mark.parametrize("env", TESTING_OBS_ENVS, ids=TESTING_OBS_ENVS_IDS)
def test_normalize_obs_spaces(env):
    normalized_env = NormalizeObservationV0(env)

    obs, _ = normalized_env.reset()
    for _ in range(100):
        assert obs in normalized_env.observation_space

        obs, _, _, _, _ = normalized_env.step(env.action_space.sample())
```

**Comment by KexianShen** [comment]
The solution works. I will keep this issue open until a stable release. Thanks.

**Comment by Kallinteris-Andreas** [comment]
@pseudo-rnd-thoughts can you close if fixed

**Comment by pseudo-rnd-thoughts** [comment]
Fixed in #978
